Containerfile.base

```diff
@@ -0,0 +1,24 @@
+FROM public.ecr.aws/docker/library/ruby:3.0.2-slim-bullseye AS ruby-upstream
+
+RUN apt update && \
+    apt install -y \
+        curl \
+        gnupg2 \
+        && \
+    curl -fsSL https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+    curl -fsSL https://deb.nodesource.com/setup_16.x | bash - && \
+    echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
+    apt update && \
+    apt install -y \
+        imagemagick \
```

Comment:
IM's inclusion here is because the `quay.io/forem/ruby` this image is replacing, installed it as part of the base image. I'm open to the idea that this actually belongs in the app-specific containerfile - honestly I suspect I'll be moving various dependencies between the two as time goes on (as I prepare the app-specific containerfile's rebase onto this image, I'm discovering dependencies that probably need moved into the base, too)